superbench/benchmarks/docker_benchmarks/docker_base.py

```diff
@@ -36,24 +44,87 @@ def _preprocess(self):
         Return:
             True if _preprocess() succeed.
         """
-        return super()._preprocess()
+        if not super()._preprocess():
+            return False
+
+        if self._image is None:
+            self._result.set_return_code(ReturnCode.DOCKERBENCHMARK_IMAGE_NOT_SET)
+            logger.error('The image url is not set - benchmark: {}.'.format(self._name))
+            return False
+
+        if self._container is None:
+            self._result.set_return_code(ReturnCode.DOCKERBENCHMARK_CONTAINER_NOT_SET)
+            logger.error('The container name is not set - benchmark: {}.'.format(self._name))
+            return False
+
+        output = run_command('docker pull {}'.format(self._image))
+        if output.returncode != 0:
+            self._result.set_return_code(ReturnCode.DOCKERBENCHMARK_IMAGE_PULL_FAILURE)
+            logger.error(
+                'Docker benchmark pull image failed - benchmark: {}, error message: {}.'.format(
+                    self._name, output.stdout
+                )
+            )
+            return False
+
+        return True
+
+    def _postprocess(self):
+        """Postprocess/cleanup operations after the benchmarking.
+
+        Return:
+            True if _postprocess() succeed.
+        """
+        rm_containers = 'docker rm $(docker stop {}))'.format(self._container)
+        run_command(rm_containers)
+
+        rm_image = 'docker rmi {}'.format(self._image)
+        run_command(rm_image)
+
+        return True
 
-    @abstractmethod
     def _benchmark(self):
-        """Implementation for benchmarking."""
-        pass
+        """Implementation for benchmarking.
 
-    def _process_raw_result(self, raw_output):
+        Return:
+            True if run benchmark successfully.
+        """
+        for cmd_idx in range(len(self._commands)):
```

Comment:
will docker benchmark support commands?
I think the best way is using an entrypoint, so `docker run` will execute also contents directly

Reply:
Yes. support commands. We can't guarantee the launch way for the images from third-party, just leave the extendibility to the contributor. For ONNX image, it has only one command, which looks like:
```
docker run -it --rm \
  --privileged \
  --ipc=host \
  --device=/dev/kfd \
  --device=/dev/dri \
  --group-add video \
  --cap-add=SYS_PTRACE \
  --security-opt seccomp=unconfined \
  -v /myhost/working/path:/workspace/ \
  -v /myhost/working/path/logs:/logs/ \
  --workdir /workspace/ \
  sukha.azurecr.io/onnxruntime/rocm4.2:benchmark
```

Reply:
I think for docker benchmark, the input only contains: `image_uri` and `entrypoint`? e.g.,
```yaml
ort_xxx_docker:
  image_uri: onnxruntime/rocm4.2:benchmark
  entrypoint: python ...
```
the docker can be run as privileged and gpus mounted, but why do we need to mount any path from host?
if you let user to decide whole docker command, it's real dangerous:
```sh
docker run -it --privileged -v /:/ onnxruntime/rocm4.2:benchmark rm -rf /
```

Reply:
Agree, we should have some principles for docker image provided to us. Like 1. can't rely on host path. 2. print the result by stdout, etc.